hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorServiceTest.java

```diff
@@ -0,0 +1,618 @@
+
+
+        ScheduledTaskHandler handler = first.getHandler();
+        int expectedPartition = instances[0].getPartitionService()
+                                            .getPartition(key)
+                                            .getPartitionId();
+        assertEquals(expectedPartition, handler.getPartitionId());
+        assertEquals(25, first.get(), 0);
+    }
+
+    @Test
+    public void scheduleOnKeyOwnerWithRepetition()
+            throws InterruptedException {
+        String key = "TestKey";
+
+        HazelcastInstance[] instances = createClusterWithCount(2);
+        IScheduledExecutorService executorService = instances[0].getScheduledExecutorService("s");
+
+        ICountDownLatch latch = instances[0].getCountDownLatch("latch");
+        latch.trySetCount(5);
+
+        IScheduledFuture future = executorService.scheduleOnKeyOwnerWithRepetition(
+                new ICountdownLatchRunnableTask("latch", instances[0]), key,
+                0, 1, TimeUnit.SECONDS);
+
+        ScheduledTaskHandler handler = future.getHandler();
+        int expectedPartition = instances[0].getPartitionService()
+                                            .getPartition(key)
+                                            .getPartitionId();
+
+        assertEquals(expectedPartition, handler.getPartitionId());
+
+        latch.await(10, TimeUnit.SECONDS);
+        assertEquals(0, latch.getCount());
+    }
+
+    @Test
+    public void getScheduled() {
+        int delay = 1;
+        String taskName = "Test";
+
+        HazelcastInstance[] instances = createClusterWithCount(2);
+        IScheduledExecutorService executorService = instances[0].getScheduledExecutorService("s");
+        IScheduledFuture<Double> first = executorService.schedule(
+                named(taskName, new PlainCallableTask()), delay, TimeUnit.SECONDS);
+
+        ScheduledTaskHandler handler = first.getHandler();
+        IScheduledFuture<Double> copy = executorService.getScheduledFuture(handler);
+
+        assertEquals(first, copy);
+    }
+
+    @Test
+    public void getAllScheduled()
+            throws ExecutionException, InterruptedException {
+
+        HazelcastInstance[] instances = createClusterWithCount(3);
+        IScheduledExecutorService s = instances[0].getScheduledExecutorService("s");
+        s.scheduleOnAllMembers(new PlainCallableTask(), 0, TimeUnit.SECONDS);
+
+        Set<Member> members = instances[0].getCluster().getMembers();
+        Map<Member, List<IScheduledFuture<Double>>> allScheduled = s.getAllScheduledFutures();
+
+        assertEquals(members.size(), allScheduled.size());
+
+        for (Member member : members) {
+            assertEquals(1, allScheduled.get(member).size());
+            assertEquals(25.0, allScheduled.get(member).get(0).get(), 0);
+        }
+    }
+
+    public HazelcastInstance[] createClusterWithCount(int count) {
+        HazelcastInstance[] instances = new HazelcastInstance[count];
```

Comment:
you don't have to create instance array explicitly, check `factory.newInstances()` method